packages/core/src/components/document-component/document-component.tsx

```diff
@@ -326,11 +326,16 @@ export class DocumentComponent implements ComponentInterface {
                     </dso-label>
                   )}
                   {this.annotated && (
-                    <dso-annotation-button
-                      identifier="expandable"
-                      open={this.openAnnotation}
-                      onDsoClick={(e) => this.dsoAnnotationToggle.emit({ originalEvent: e })}
-                    ></dso-annotation-button>
+                    <button
+                      type="button"
+                      class="dso-tertiary"
+                      aria-controls="expandable"
```

Comment:
Ik heb iets over het hoofd gezien.

Het element (met het id `expandable`) waar dit attribuut naar verwijst bestaat niet in deze shadow root. Dat deed het in de oude situatie ook niet.

Annotation Button en Annotation Output waren componenten zonder shadow root, er stond ook een melding bij dat aangaf dat dit bewust was gedaan vanwege a11y redenen.

Ik vermoed dat `<dso-annotation-button>` wel ooit eens verwees naar `<dso-annotation-output>`, maar dat het bij 'n refactor eens verloren is gegaan.

Maar ik weet nu niet zo goed wat we wel moeten doen.

> [!NOTE]
> `aria-expanded` en `aria-controls` worden op de control (deze button) gezet. `aria-controls` verwijst naar het element wat wordt bestuurd.
> 
> **Hoe verhoudt `aria-controls` zich tot een element wat wordt verwijderd (uit het DOM)? @timveld @hansgrimm ?**
> 
> Bijvoorbeeld: een button toggled een menu wat bij een gesloten toestand zich niet in het DOM bevindt. Het menu is pas in het DOM aanwezig als het wordt getoond.

Mocht blijken dat `aria-expanded` en `aria-controls` in deze situatie nodig zijn, dan moeten ze in hetzelfde DOM leven. Het is nog niet mogelijk om cross-shadow-root referenties naar elementen te leggen, zie "leesvoer" onderaan deze comment.

Ik houd er rekening mee dat we Panel in Document Component moeten plaatsen, en niet als slottable element meegeven.

We kunnen in Document Component natuurlijk een wrapping `<div id="expandable">` om `<slot name="annotation">` zetten, maar ik denk dat het vanuit de functionele eisen van Document Component meer aansluit om Panel uberhaupt in Document Component te zetten. UX wilt dat daar altijd een Panel staat.

@iterox wat zou jij hiervan vinden?

Leesvoer:
* https://alice.pages.igalia.com/blog/how-shadow-dom-and-accessibility-are-in-conflict/
* https://wicg.github.io/aom/aria-reflection-explainer.html

Reply:
Off the top of my head, vanuit de zonnige achtertuin: als je een button hebt die met een `aria-controls` verwijst naar een ander element binnen de DOM, en dat laatste element wordt verwijderd, waarom heb je die button dan nog?

Reply:
Die button is de toggle button. De vraag is, hoe ga je om met aria-controls op `<button>` als `<ul>` niet in het DOM is als het menu niet wordt getoond:

```html
<!-- ✅ -->
<button type="button" aria-controls="menu" aria-expanded="true">Verberg menu</button>
<ul id="menu">
  [..]
</ul>

<!-- ❓ -->
<button type="button">Toon menu</button>
```

Reply:
"ARIA attributes which refer to other elements by ID should refer to elements which exist in the DOM." 
https://stackoverflow.com/questions/38384048/what-exactly-does-aria-controls-do-for-the-user-how-is-it-affected-by-ajax-usag (8 jaar oud, maar m.i. nog wel geldig (zo hard gaan ze niet bij de WCAG)).